class-factory/native/src/lib.rs

```diff
@@ -84,10 +84,12 @@ declare_types! {
         method log(mut cx) {
             let log = cx.argument::<JsString>(0)?;
             let this = cx.this();
-            let guard = cx.lock();
+            {
```

Comment:
Why the additional scoping here? This shouldn't be necessary with non-lexical lifetimes. Can we remove it? It might obscure intent.

Reply:
Oh, the newest version of rust can compile it without needing the additional scoping, I'll revert this change.